tests/PersistentSQLTest.php

```diff
@@ -51,4 +51,83 @@ public function testLoadArray()
         $this->assertEquals('Smith', $m['surname']);
     }
 
+    public function testPersistenceInsert()
+    {
+        $a = [
+            'user'=>[
+                1=>['name'=>'John', 'surname'=>'Smith'],
+                2=>['name'=>'Sarah', 'surname'=>'Jones'],
+            ]
+        ];
+
+        $p = new Persistence_SQL('sqlite::memory:');
+
+        $p->connection->expr('drop table if exists user')->execute();
+        $p->connection->expr('create table user(id integer primary key autoincrement, name varchar(255), surname varchar(255))')->execute();
+
+        $m = new Model($p, 'user');
+        $m->addField('name');
+        $m->addField('surname');
+
+
+        $ids = [];
+        foreach($a['user'] as $id=>$row){
+            $ids[] = $p->insert($m, $row);
+        }
+
+        $m->load($ids[0]);
+        $this->assertEquals('John', $m['name']);
+
+        $m->load($ids[1]);
+        $this->assertEquals('Jones', $m['surname']);
+        $m['surname'] = 'Smith';
+        $m->save();
+
+        $m->load($ids[0]);
+        $this->assertEquals('John', $m['name']);
+
+        $m->load($ids[1]);
+        $this->assertEquals('Smith', $m['surname']);
+    }
+
+    public function testPersistenceDelete()
+    {
+        $a = [
+            'user'=>[
+                1=>['name'=>'John', 'surname'=>'Smith'],
+                2=>['name'=>'Sarah', 'surname'=>'Jones'],
+            ]
+        ];
+
+        $p = new Persistence_SQL('sqlite::memory:');
+
+        $p->connection->expr('drop table if exists user')->execute();
+        $p->connection->expr('create table user(id integer primary key autoincrement, name varchar(255), surname varchar(255))')->execute();
```

Comment:
Is this a proper generic SQL syntax. AFAIK primary key cannot be set like this in all SQL implementations